### ISSUE TITLE
Show tags per bible books

### ISSUE THREAD

**Issue by debritto** [ezra-bible-app/ezra-bible-app#527]
Hi guys! When we choose some tags to be shown, the App shows a cool chart summarizing how many tags were marked by Bible book, as we can see here:

![tags_summary](https://user-images.githubusercontent.com/9845780/142670459-0f2530ac-88d7-4e21-bd1e-54ec3c53f22e.png)

What do you think about the possibility to show Bible books occurrences for every selected tag. This chart shows how every tag "spread" throughout the Bible.

Please, see an example below from a spreadsheet that I made to show this kind of information:

![tags_summary_categorized](https://user-images.githubusercontent.com/9845780/142671043-c6924651-44ca-43a2-ace4-1e9380781c32.png)

Note that the numbers in the yellow boxes shows  occurrences for every tag.
Warm regards from Brazil!

**Comment by tobias-klein** [comment]
Good idea! So, this is obviously most useful when having a list of tagged verses based on a selection of multiple tags.
Here are some questions:

1. Would you still show this even if only one tag is selected? Or hide/not generate the chart in that case?
2. Would you show it additionally to the existing chart? Should the chart show up immediately or maybe after pressing a button?
3. May this chart also make sense on book level in case of larger books? (Think of Isaiah, Psalms, Revelation, the Gospels ...)

**Comment by debritto** [comment]
1.Would you still show this even if only one tag is selected? Or hide/not generate the chart in that case?
- In my case I show all the tags, even if they are do not used. In this way I could learn how topic I still need to research in the Bible. As a matter of fact, this table was created to defy my students to research how a topic is approached by The Bible as a whole. I want to challenge them to read all the Bible in order to understand their personal study topics.

2.Would you show it additionally to the existing chart? Should the chart show up immediately or maybe after pressing a button?
- May a button below the current chart could be used. Something like "Show Occurrences Per Tag" or "Advanced Chart".
3.May this chart also make sense on book level in case of larger books? (Think of Isaiah, Psalms, Revelation, the Gospels ...)
- I think so, sometimes is important to see how a topic spread through a specific book. Good idea. In this case, will be interesting to show the book emphasized with all its verses or chapters.
- Warm regards

**Comment by tobias-klein** [comment]
Hi @debritto,

So I played a bit with an implementation for this feature.
This table would get quite large if it shows all books of the whole Bible. So in my test implementation I limited it to the books that are part of the data.

How do you like this?

![image](https://user-images.githubusercontent.com/17739895/199295979-0332cd4b-a43f-4a47-9848-08f2301a8e83.png)

**Comment by debritto** [comment]
Hi, @tobias-klein ! Awesome. That is excellent. If you could use some CSS style to make it aesthetically better will be perfect. I say this because I think that this kind of graphics could make Ezra Bible better than other software packages. But, simple as that, is already very useful. Do you think that is possible to copy and paste it?

**Comment by tobias-klein** [comment]
Thank you for the quick feedback @debritto.

I added the alternating background rows and adjusted the style of the headers, so that it matches the chart above. Any other style suggestions?

Regarding copy & paste: At last, I had issues with structures other than plain text. However, I will check once more. I think you had a similar request regarding the compare translations feature, right.

![image](https://user-images.githubusercontent.com/17739895/199421764-9efda2d2-d824-4034-97d5-dfd9c9e310ce.png)

**Comment by tobias-klein** [comment]
I managed to get the copy & paste implemented!

There is a button below the table. The table is copied as html. I tried to paste it in LibreOffice Writer and Calc and it worked!

![image](https://user-images.githubusercontent.com/17739895/199446371-235a48e2-91d0-496a-b1e7-bbcb762afd74.png)

**Comment by debritto** [comment]
Hi @tobias-klein !
You have made an amazing job! The copy to clipboard idea is very good. I had a student that will be defending his undergraduate dissertation in Theology, and he is using exactly this graphic. This will be very useful.

Talking about style. I think that information needed is already on the chart. But I think that we can add color information, so the user can get more insights with the information. There is a chart type called "heatmap" that makes use of this idea with this kind of information. Please, take a look here for a very good example:  [Heatmap Example](https://www.highcharts.com/demo/heatmap)

![image](https://user-images.githubusercontent.com/9845780/199529495-bdc39269-b001-4178-a36b-fcb72a2c7d6e.png)


Note that the colors changes, from light to darken colors, according to the quantity in the cells.
I think that this kind of details could make Ezra Bible app excels between the other applications.

**Comment by tobias-klein** [comment]
Hi @debritto,

Yes, a heatmap could be interesting!

How about this?

<img width="585" alt="image" src="https://user-images.githubusercontent.com/17739895/199559221-237deb26-850d-41f5-8163-0e2d7b382c5f.png">

I have put the values in 20% clusters and then mapped them to a color range of five clusters.

**Comment by debritto** [comment]
I love it! Could not be better. Congratulations 😉

**Comment by debritto** [comment]
May I could test it here? I'am using Linux version

**Comment by tobias-klein** [comment]
@debritto Can you try one of the builds available [here](https://github.com/ezra-bible-app/ezra-bible-app/actions/runs/3380485400) (scroll all the way down).

**Comment by tobias-klein** [comment]
@debritto Any feedback before I consider this done? :)

**Comment by debritto** [comment]
No. It is perfect!

**Comment by tobias-klein** [comment]
Thanks! This enhancement will be part of 1.9.0 - this year's Christmas release! :)
I just merged the changes to master.
